server/users.go

```diff
@@ -27,15 +28,26 @@ func (r *userRequest) ValidCreate() error {
 	if r.Scheme == "" {
 		return fmt.Errorf("Scheme required on Chronograf User request body")
 	}
-	return nil
+	return r.ValidRoles()
 }
 
 // TODO: Provide detailed error message
 // TODO: Reconsider what fields should actually be updateable once this is more robust
 func (r *userRequest) ValidUpdate() error {
-	if r.Name == "" && r.Provider == "" && r.Scheme == "" {
+	if r.Name == "" && r.Provider == "" && r.Scheme == "" && r.Roles == nil {
 		return fmt.Errorf("No fields to update")
 	}
+	return r.ValidRoles()
+}
+
+func (r *userRequest) ValidRoles() error {
+	if len(r.Roles) > 0 {
+		for _, r := range r.Roles {
+			if r != chronograf.ViewerRoleName && r != chronograf.EditorRoleName && r != chronograf.AdminRoleName {
+				return fmt.Errorf("Invalid role assignment '%s' on Chronograf User request body", r)
```

Comment:
I think it is more like : `Unknown role %s.  Valid roles are "Viewer", "Editor", "Admin", "SuperAdmin"`

Reply:
👍